### PR TITLE
MGMT-4179 Fix installer handling of ignitions to support multiple ver…

### PR DESCRIPTION
**Pull request by tsorya**
…sions - sno mode

### CODE REVIEW

File: src/ignition/ignition.go [hunk 1]
```diff
@@ -26,20 +30,34 @@ func NewIgnition() *ignition {
...
 	if err != nil {
-		return nil, errors.Wrap(err, "error parsing ignition")
+		return nil, err
 	}
-	return &config, nil
+	return config, nil
+}
+
+// ParseIgnitionFile reads an ignition config from a given path on disk
+func (i *ignition) parseToLatest(content []byte) (*types.Config, error) {
```
Comment (eranco74) [inline src/ignition/ignition.go:46]: Why not use the same code as the service (where we force 3.2...)?
I think you can actually use the same code as a library...

File: src/installer/installer.go [hunk 2]
```diff
@@ -153,7 +151,7 @@ func (i *installer) updateSingleNodeIgnition(singleNodeIgnitionPath string) erro
 	}
 	// TODO: update this once we can get the full host specific overrides we have in the ignition
 	// Remove the Config part since we only want the rest of the overrides
-	hostConfig.Ignition.Config = config31types.IgnitionConfig{}
+	hostConfig.Ignition.Config = ignition.EmptyIgnitionConfig
```
Comment (eranco74) [inline src/installer/installer.go:154]: Hmmm, seems like this should be the master ignition we get from the service no?

File: src/ignition/ignition.go (hunk 1 above)
Comment (tsorya) [reply src/ignition/ignition.go:46]: it is the same code but it gives us an option just to change import and not go through all variables

Comment (tsorya) [reply src/ignition/ignition.go:46]: so it is just naming thing

Comment (eranco74) [reply src/ignition/ignition.go:46]: Ok

**Review by eranco74 — COMMENTED** [review]
/lgtm

File: src/installer/installer.go (hunk 2 above)
Comment (tsorya) [reply src/installer/installer.go:154]: why?

Comment (tsorya) [reply src/installer/installer.go:154]: if i understand right the comment, we just remove all the host specific ignition part

**Pull request merged**